src/util/sighash.rs

```diff
@@ -198,7 +199,7 @@ impl<'u> Prevouts<'u> {
         Ok(())
     }
 
-    fn get_all(&self) -> Result<&[TxOut], Error> {
+    fn get_all(&self) -> Result<&&[T], Error> {
```

Comment:
Why is there double borrow and how come it even works?!

Reply:
This was just compiler error chasing and something confused me too. If I call `self.clone()` in `get_all`, I would expect something of type Prevouts, but I get a type `&Prevouts`. Maybe you can help me here :) 

I would guess that the double ref works because of auto-deref.

Reply:
Oh, now I realized how it can even work. I see no reason why having one ref here and dereferencing `prevouts` in match arm with `*` wouldn't work.

Reply:
That should work, trying it out.